tools/scripts/script.ps1

```diff
@@ -23,161 +23,168 @@ param(
-    Expand-Archive -Path $destinationZipFile -DestinationPath $destinationUnzippedFolder -Force -ErrorAction SilentlyContinue
+    $tempFilesDirectory = Join-Path $destinationDirectory "temp"
+    if(!(Test-Path -path $tempFilesDirectory)) {
+        New-Item -Path $tempFilesDirectory -ItemType Directory -Force | Out-Null
+    }
+    Write-Host "[Downloading the latest AIShell PowerShell Module to $tempFilesDirectory ...]`n" -ForegroundColor Green
+    Invoke-WebRequest -Uri $moduleUrl -OutFile $tempFilesDirectory
 
-    # check os
+    # Download OS specific AIShell build
     $osPlatform = ""
+    $destinationArchiveFile = ""
     if ($IsWindows) {
         $osPlatform = "Windows"
+        $destinationArchiveFile = Join-Path $tempFilesDirectory "AIShell.zip"
     } 
     elseif ($IsMacOS) {
         $osPlatform = "MacOS"
+        $destinationArchiveFile = Join-Path $tempFilesDirectory "AIShell.tar.gz"
     }
-    else {
-        Write-Error "Sorry, AIShell currently only supports Windows and MacOS"
-        exit 1
-    }
-
-    # check architecture
-    $osArchitecture = [System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture
-    if ($osArchitecture -ne "X64" -and $osArchitecture -ne "Arm64") {
-        Write-Error "Sorry, AIShell currently only supports x64 and arm64 architectures"
-        exit 1
-    }
-
-    # Download OS specific AIShell build
-    $downloadUrl = ""
-    if ($osPlatform -eq "Windows" -and $osArchitecture -eq "X64"){
-        $downloadUrl = $winX64Url
-    } 
-    elseif ($osPlatform -eq "Windows" -and $osArchitecture -eq "Arm64"){
-        $downloadUrl = $winArm64Url
-    } 
-    elseif ($osPlatform -eq "MacOS" -and $osArchitecture -eq "X64"){
-        $downloadUrl = $osxX64Url
-    } 
-    elseif ($osPlatform -eq "MacOS" -and $osArchitecture -eq "Arm64"){
-        $downloadUrl = $osxArm64Url
-    } 
-    $destinationZipFile = Join-Path $destinationDirectory "AIShell.zip"
-    Write-Host "[Downloading the latest AIShell for $osPlatform $osArchitecture to $destinationZipFile ...]`n" -ForegroundColor Green
-    Invoke-WebRequest -Uri $downloadUrl -OutFile $destinationZipFile
-    # $ProgressPreference = 'Continue'
+    Write-Host "[Downloading the latest AIShell for $osPlatform $osArchitecture to $destinationArchiveFile ...]`n" -ForegroundColor Green
+    Invoke-WebRequest -Uri $buildUrl -OutFile $destinationArchiveFile
     
     # Unzip AIShell
-    $destinationUnzippedFolder = Join-Path $destinationDirectory "AIShell"
-    Write-Host "[Unzipping AIShell to $destinationUnzippedFolder ...]" -ForegroundColor Green
-    Expand-Archive -Path $destinationZipFile -DestinationPath $destinationUnzippedFolder -Force
+    Write-Host "[Unzipping AIShell to $destinationDirectory ...]" -ForegroundColor Green
+    Unblock-File -Path $destinationArchiveFile
+    Expand-Archive -Path $destinationArchiveFile -DestinationPath $destinationDirectory -Force
```

Comment:
Did you actually check the packages I shared with you? The macOS package is a `.tar.gz` file, and `Expand-Archive` doesn't work for it.